src/vsr/replica.zig

```diff
@@ -2272,75 +2337,69 @@ pub fn ReplicaType(
                 ).?,
             );
 
-            const view_headers = start_view_message_headers(message);
-            assert(view_headers[0].op == message.header.op);
-            assert(view_headers[0].op >= view_headers[view_headers.len - 1].op);
+            if (!vsr.Checkpoint.durable(self.op_checkpoint_next(), message.header.commit_max)) {
+                return false;
+            }
 
-            if (vsr.Checkpoint.durable(self.op_checkpoint_next(), message.header.commit_max) and (
-            //  Cluster is at least two checkpoints ahead. Although SV's checkpoint is not
-            //  guaranteed to be durable on a quorum of replicas, it is safe to sync to it, because
-            //  prepares in this replica's WAL are no longer needed.
-                vsr.Checkpoint.durable(self.op_checkpoint_next() +
-                constants.vsr_checkpoint_ops, message.header.commit_max) or
-                // Cluster is on the next checkpoint, and that checkpoint is durable and is safe to
-                // sync to. Try to optimistically avoid state sync and prefer WAL repair, unless
-                // there's evidence that the repair can't be completed.
-                (self.syncing == .idle and self.repair_stuck()) or
-                // Completing previously starting state sync.
-                self.syncing == .awaiting_checkpoint))
-            {
-                // State sync: at this point, we know we want to replace our checkpoint
-                // with the one from this SV.
+            // Cluster is at least two checkpoints ahead. Although SV's checkpoint is not guaranteed
+            // be durable on a quorum of replicas, it is safe to sync to it, because prepares in
+            // this replica's WAL are no longer needed.
+            const far_behind = vsr.Checkpoint.durable(self.op_checkpoint_next() +
+                constants.vsr_checkpoint_ops, message.header.commit_max);
+            // Cluster is on the next checkpoint, and that checkpoint is durable and is safe to
+            // sync to. Try to optimistically avoid state sync and prefer WAL repair, unless
+            // there's evidence that the repair can't be completed.
+            const likely_stuck = self.syncing == .idle and self.repair_stuck();
```

Comment:
This I think would be the right place to implement that other idea where a replica which syncs tables triggers state sync more aggressively, to avoid the situation where it waits for an entire `repair_sync_timeout` before triggering the sync, all while the cluster is making progress in the new checkpoint already.

Reply:
It is the right place, but I think I will include the actual change in https://github.com/tigerbeetle/tigerbeetle/pull/2621 instead. The reason is that as of that PR, `sync_tables` will get set earlier (immediately after we sync the checkpoint, rather than after state-machine open), so checking `sync_tables != null` can be used to answer the "are we already busy syncing" question.